compaction/compaction.cc

```diff
@@ -635,8 +635,8 @@ class compaction {
             // sstables should never be shared with other shards at this point.
             assert(dht::shard_of(*_schema, dk.token()) == this_shard_id());
 #endif
-
-            if (!_owned_ranges_checker->belongs_to_current_node(dk.token())) {
+            auto zis = const_cast<compaction*>(this);
+            if (!zis->_owned_ranges_checker->belongs_to_current_node(dk.token())) {
                 log_trace("Token {} does not belong to this node, skipping", dk.token());
```

Comment:
I don't think this is any better than mutable keyword, which states that a data member can be modified even when this is const. owned_ranges_checker is not stateless, therefore it makes sense to make it mutable.

Reply:
Agree. But why is owned_ranges_checker not stateless? It seems odd for something named "checker".

Reply:
It increments an iterator to provide average O(1) for each step.

```
class incremental_owned_ranges_checker {
    const dht::token_range_vector& _sorted_owned_ranges;
    mutable dht::token_range_vector::const_iterator _it;
```

with iterator itself made mutable by the class, it seems to me we just need to make incremental_owned_ranges_checker::belongs_to_current_node() const qualified and then compaction can remove its mutable for the checker. better, no?

Reply:
Yes. The checker is logically const (as the name suggests), the mutable _it is an implementation detail.

Reply:
I submitted https://github.com/scylladb/scylladb/pull/12634 to mark belongs_to_current_node as const so we can remove the mutable definition in cleanup_compaction